docs/src/main/sphinx/connector/faker.md

```diff
@@ -0,0 +1,225 @@
+and [available providers](https://www.datafaker.net/documentation/providers/).
+
+To test a generator expression, without having to recreate the table, use the
+`random_string(expression)` function:
+
+```sql
+SELECT random_string('#{Name.first_name}')
+```
+
+### Non-character types
+
+Faker supports the following non-character types:
+
+- `BIGINT`
+- `INTEGER` or `INT`
+- `SMALLINT`
+- `TINYINT`
+- `BOOLEAN`
+- `DATE`
+- `DECIMAL`
+- `REAL`
+- `DOUBLE`
+- `INTERVAL DAY TO SECOND`
+- `INTERVAL YEAR TO MONTH`
+- `TIMESTAMP` and `TIMESTAMP(P)`
+- `TIMESTAMP WITH TIME ZONE` and `TIMESTAMP(P) WITH TIME ZONE`
+- `TIME` and `TIME(P)`
+- `TIME WITH TIME ZONE` and `TIME(P) WITH TIME ZONE`
+- `IPADDRESS`
+- `UUID`
+
+Generator expressions cannot be used for non-character-based columns. To limit
+their data range, specify constraints in the `WHERE` clause.
+
+### Unsupported types
+
+Faker does not support the following data types:
+
+- structural types: `ARRAY`, `MAP`, `ROW`
+- `JSON`
+- Geometry
+- HyperLogLog and all digest types
+
+To generate data using these complex types, data from column of primitive types
+can be combined, like in the following example.
+
+```sql
+CREATE TABLE faker.default.prices (
+  currency VARCHAR NOT NULL WITH (generator = '#{Currency.code}'),
+  price DECIMAL(8,2) NOT NULL
+);
+
+SELECT JSON_OBJECT(KEY currency VALUE price) AS complex
+FROM faker.default.prices
+WHERE price > 0
+LIMIT 3;
+```
+
+Executing these queries should return data structured like this:
+```
+      complex
+-------------------
+ {"TTD":924657.82}
+ {"MRO":968292.49}
+ {"LTL":357773.63}
+(3 rows)
+```
+
+### Number of generated rows
+
+By default, the connector generates 1000 rows for every table. To control how
+many rows are generated for a table, use the `LIMIT` clause in the query. A
+default limit can be set using the `default_limit` table, or schema property or
+in the connector configuration file, using the `faker.default-limit` property.
+
+### Null values
+
+For columns without the `NOT NULL` constraint, null values are generated using
+the default probability of 50%. It can be modified using the `null_probability`
+property set for a column, table, or schema. The default value of 0.5 can be
+also modified in the connector configuration file, by using the
+`faker.null-probability` property.
+
+(faker-type-mapping)=
+## Type mapping
+
+The Faker connector generates data itself, so no mapping is required.
+
+(faker-sql-support)=
+## SQL support
+
+The connector provides {ref}`globally available <sql-globally-available>` and
+{ref}`read operation <sql-read-operations>` statements to generate data.
+
+To define the schema for generating data, it supports the following features:
+
+- {doc}`/sql/create-table`
+- {doc}`/sql/create-table-as`
+- {doc}`/sql/drop-table`
+- {doc}`/sql/create-schema`
+- {doc}`/sql/drop-schema`
+
+## Limitations
+
+- Generated data is not deterministic. There is no way to specify a seed for
```

Comment:
Do we maybe need to explain that since the implications I kinda odd .. if I run a query that analyses data in a catalog with a faker table twice .. the results will be different each time .. for example you have age colum and you do a count of people under 30 .. that will change if you run the query twice.. right? 

the only way to avoid that is to use the connector once so the data is inserted in a fixed table in another catalog..  

This is not obvious so i think we need to explain that

Reply:
Yes, same query returns different random data every time. That's exactly why I added this section. I added more detail, PTAL.

The connector is installed in the free Trino cluster available at `trino.rocks`, so you can test it yourself:
```
% trino trino.rocks
trino> select * from faker.default.primitives limit 10;
 currency |   price    
----------+------------
 RWF      |  554344.83 
 BHD      |  423285.44 
 HNL      |  444472.14 
 ETB      |   62409.52 
 MDL      |   29279.57 
 MOP      |  341507.29 
 FJD      | -698746.15 
 CNY      |  791697.12 
 MZN      | -809493.51 
 BWP      | -344078.60 
(10 rows)

trino> select * from faker.default.primitives limit 10;
 currency |   price    
----------+------------
 OMR      |  -75039.71 
 BBD      | -925266.91 
 HRK      |  541090.25 
 JPY      |  119664.79 
 BIF      | -196237.23 
 CNY      |  -91127.24 
 RON      |  437617.12 
 GBP      | -773280.08 
 TMT      |  203127.32 
 XPF      |   69625.73 
(10 rows)
```

Reply:
In the intro, we mention using this connector to populate a different data source.

Reply:
I'm not sure how to describe it more - as I said in the previous comment, I already added a mention of this at the top of this doc.